File: ruby/lesson 6/post_order.rb
# https://leetcode.com/problems/binary-tree-postorder-traversal/

def postorder_traversal(root)
    result = []
    postorder(root, result)
    result
end

def postorder(node, result)
    return if node.nil?

    postorder(node.left, result)
    postorder(node.right, result)
    result.push(node.val)
end

File: python/lesson 6/post_order.py
# https://leetcode.com/problems/binary-tree-postorder-traversal/

class Solution:
    def postorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        result = []
        self.postorder(root, result)
        return result
    
    def postorder(self, node: Optional[TreeNode], result: List[int]):
        if node == None: 
            return
        self.postorder(node.left, result)
        self.postorder(node.right, result)
        result.append(node.val)

